pkg/k8s/k8s_pod_probes.go

```diff
@@ -0,0 +1,66 @@
+package k8s
+
+import "fmt"
+
+// To keep the structs minimal, everything that is not needed right now is commented out
+// If you need it, uncomment it and add the needed fields
+// This way we can keep the structs minimal and add more without a breaking change
+
+// Probe is a structure to hold the probes for an instance
+type Probe struct {
+	//Exec                          *ExecAction
+	HTTPGet             *HTTPGetAction   // Only one action is allowed, deployment will fail otherwise
+	TCPSocket           *TCPSocketAction // Only one action is allowed, deployment will fail otherwise
+	InitialDelaySeconds int32            // Optional
+	//TerminationGracePeriodSeconds int64
+	//PeriodSeconds                 int32
+	//TimeoutSeconds                int32
+	//FailureThreshold              int32
+	//SuccessThreshold              int32
+	//GRPC                          *GRPCAction
+}
+
+func (p *Probe) String() string {
+	if p.HTTPGet != nil {
+		return fmt.Sprintf("HTTPGet: %s, InitialDelaySeconds: %d", p.HTTPGet.String(), p.InitialDelaySeconds)
+	}
+	if p.TCPSocket != nil {
+		return fmt.Sprintf("TCPSocket: %s, InitialDelaySeconds: %d", p.TCPSocket.String(), p.InitialDelaySeconds)
+	}
+	return ""
```

Comment:
Can we give a hint why empty string is good here?

Reply:
compiler I guess

Reply:
What would you propose to return here?

Reply:
I would return something meaningful like 'Probing TCP and HTTP sockets has recorded zero delays'. 

I don't see "" being useful for a human being. And for code-base further logic, we should refer to int InitialDelaySeconds rather then String()